### PR TITLE
Update README for new cache location

### LINKED ISSUE #2216: Change cache location on Mac/Windows
Currently, the base image cache is located at:
* Mac: `$HOME/Library/Application Support/google-cloud-tools-java/jib/`
* Win: `%LOCALAPPDATA/google-cloud-tools-java/jib/`

It may be a good idea to change these to:
* Mac: `$HOME/Library/Caches/Google/Jib/`
* Win: `%LOCALAPPDATA/Google/Jib/Cache/`

We may need to come up with a migration strategy so we don't break users who rely heavily on the cache (e.g. offline users).

Checklist:
- [x] Code (#2218)
- [x] Docs (#2231)

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #2216.

### CODE REVIEW

File: jib-gradle-plugin/README.md [hunk 1]
```diff
@@ -271,14 +271,16 @@ Property | Type | Default | Description
 --- | --- | --- | ---
 `jib.httpTimeout` | `int` | `20000` | HTTP connection/read timeout for registry interactions, in milliseconds. Use a value of `0` for an infinite timeout.
 `jib.useOnlyProjectCache` | `boolean` | `false` | If set to true, Jib does not share a cache between different Maven projects.
-`jib.baseImageCache` | `File` | `[user cache home]/google-cloud-tools-java/jib` | Sets the directory to use for caching base image layers. This cache can (and should) be shared between multiple images.
+`jib.baseImageCache` | `File` | `[user cache home]/google-cloud-tools-java/jib`\*\*\* | Sets the directory to use for caching base image layers. This cache can (and should) be shared between multiple images.
```
Comment (chanseokoh) [inline jib-gradle-plugin/README.md]: Hmm... it is not clear what `[user cache home]` is. I guess, for example, it is either `$HOME/.cache/google-cloud-tools-java/jib` or `$XDG_CACHE_HOME/jib`? I actually forgot the exact path.

Comment (TadCordle) [reply jib-gradle-plugin/README.md]: Here `[user cache home]` means either `$XDG_CACHE_HOME` or `$HOME/.cache/`. Writing out of the full directory for everything gets verbose pretty quickly so I kept the shorthand. But if you think it's necessary to be more explicit I can change it.

Comment (chanseokoh) [reply jib-gradle-plugin/README.md]: I agree. How about `[platform-dependent cache root]` instead of `[user cache home]`?

Comment (TadCordle) [reply jib-gradle-plugin/README.md]: This is specifically linux; the footnote goes into other platforms. I'll change it to be clear.

File: jib-gradle-plugin/README.md [hunk 2]
```diff
@@ -271,14 +271,19 @@ Property | Type | Default | Description
 --- | --- | --- | ---
 `jib.httpTimeout` | `int` | `20000` | HTTP connection/read timeout for registry interactions, in milliseconds. Use a value of `0` for an infinite timeout.
 `jib.useOnlyProjectCache` | `boolean` | `false` | If set to true, Jib does not share a cache between different Maven projects.
-`jib.baseImageCache` | `File` | `[user cache home]/google-cloud-tools-java/jib` | Sets the directory to use for caching base image layers. This cache can (and should) be shared between multiple images.
+`jib.baseImageCache` | string | *Platform-dependent*\*\*\* | Sets the directory to use for caching base image layers. This cache can (and should) be shared between multiple images.
```
Comment (chanseokoh) [inline jib-gradle-plugin/README.md]: nit: probably we can leave `File` (although I guess `string` or anything suitable will work with Gradle)? `jib.applicationCache` just below says `File`.

**Pull request merged**